tests/snuba/api/endpoints/test_organization_events_mep.py

```diff
@@ -3207,6 +3207,60 @@ def test_transaction_user_misery(self) -> None:
             },
         }
 
+    def _setup_user_misery(self, spec: OnDemandMetricSpec) -> None:
+        events = [
+            ("one", 300),
+            ("two", 300),
+            ("one", 3000),
+            ("two", 3000),
+            ("three", 400),
+            ("four", 4000),
+        ]
+        for index, event in enumerate(events):
+            email = f"{event[0]}@example.com"
+            tags = {"user.email": email}
+            if event[1] > 300 * 4:
+                tags[constants.METRIC_SATISFACTION_TAG_KEY] = constants.METRIC_FRUSTRATED_TAG_VALUE
+            self.store_on_demand_metric(
+                index,
+                spec=spec,
+                additional_tags=tags,
+                timestamp=self.day_ago + timedelta(hours=index),
+            )
+
+    def test_on_demand_user_misery_big_number(self):
+        field = "user_misery(300)"
+        query = ""
+        spec = OnDemandMetricSpec(
+            field=field,
+            query=query,
+            spec_type=MetricSpecType.DYNAMIC_QUERY,
+        )
+        self._setup_user_misery(spec)
+
+        resp = self.do_request(
+            {
+                "field": [field],
+                "project": self.project.id,
+                "query": query,
+                "dataset": "metricsEnhanced",
+                "useOnDemandMetrics": "true",
+                "onDemandType": "dynamic_query",
+            },
+        )
+        assert resp.data == {
+            "data": [{field: user_misery_formula(miserable_users=6, unique_users=6)}],
```

Comment:
I expect `user_misery_formula(miserable_users=3, unique_users=4)` rather than this.